.github/workflows/linux_unit_tests.yml

```diff
@@ -0,0 +1,84 @@
+name: Linux Unit Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+  push:
+    branches:
+      - main
+
+jobs:
+  unit_tests:
+    name: ${{ matrix.python_version }} unit tests, Core Dependencies (${{ matrix.core_dependencies }}), Codecov (${{ matrix.codecov }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - python_version: "3.7"
+            core_dependencies: false
+            codecov: false
+          - python_version: "3.8"
+            core_dependencies: false
+            codecov: true
+          - python_version: "3.8"
+            core_dependencies: true
+            codecov: true
+          - python_version: "3.9"
+            core_dependencies: true
+            codecov: false
+    steps:
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          fetch-depth: 2
+      - name: Update apt and install Graphviz
+        run: sudo apt update && sudo apt install -y graphviz
+      - if: ${{ matrix.core_dependencies }}
+        name: Installing Core Dependencies
+        run: |
+          pip install virtualenv
+          virtualenv test_python -q
+          source test_python/bin/activate
+          pip install --upgrade pip -q
+          pip install -e . --no-dependencies
+          pip install -r core-requirements.txt
+          pip install -r test-requirements.txt
+          # "!" negates return code. exit nonzero if any of these deps are found
+          ! pip freeze | grep -E "xgboost|catboost|lightgbm|plotly|ipywidgets|category_encoders"
```

Comment:
Do you think this check will still work for GH actions instead of circleci?

Quick test would be: create a do-not-merge copy of your PR, add `pip install lightgbm` above this line and make sure the job fails on that PR.

Reply:
Failed successfully
https://github.com/alteryx/evalml/runs/2256828700?check_suite_focus=true